In [9]:
# 1. Library imports
import uvicorn
from fastapi import FastAPI
from model import IrisModel, IrisSpecies

# 2. Create app and model objects
app = FastAPI()
model = IrisModel()

# 3. Expose the prediction functionality, make a prediction from the passed
#    JSON data and return the predicted flower species with the confidence
@app.post('/predict')
def predict_species(iris: IrisSpecies):
    data = iris.dict()
    prediction, probability = model.predict_species(
        data['sepal_length'], data['sepal_width'], data['petal_length'], data['petal_width']
    )
    return {
        'prediction': prediction,
        'probability': probability
    }


#suggestion: just use the app.py instead. Notebook + FastAPI is funky.


# 4. Run the API with uvicorn
#    Will run on http://127.0.0.1:8000
if __name__ == '__main__':
    pass
    uvicorn.run(app, host='127.0.0.1', port=8000)

# Example IRIS call for prediction

In [7]:
import requests 

new_measurement = {
    'sepal_length': 5.7,
    'sepal_width': 3.1,
    'petal_length': 4.9,
    'petal_width': 2.2
}

response = requests.post('http://127.0.0.1:8000/predict', json=new_measurement)
print(response.content)


b'{"prediction":"Virginica","probability":0.93}'


#### Calling the prediction in the cmd

curl -X POST "http://127.0.0.1:8000/predict" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"sepal_length\":5.7,\"sepal_width\":3.1,\"petal_length\":4.9,\"petal_width\":2.2}"


#### returns:
{"prediction":"Virginica","probability":0.93}